abci/abci.go

```diff
@@ -103,6 +114,15 @@ func (h *ProposalHandler) PrepareProposalHandler() sdk.PrepareProposalHandler {
 						continue selectBidTxLoop
 					}
 
+					feeTx, ok := refTx.(sdk.FeeTx)
+					if !ok {
+						// Malformed bundled transaction, so we remove the bid transaction
+						// and try the next top bid.
+						txsToRemove[tmpBidTx] = struct{}{}
+						continue selectBidTxLoop
+					}
+					gasLimit += feeTx.GetGas()
```

Comment:
What happens if PrepareProposalVerifyTx fails?

Reply:
gas limit will be reset on the next iteration so this is safe